os/include/tinyara/mm/mm.h

```diff
@@ -258,7 +258,7 @@ typedef size_t mmsize_t;
 /* This macro updates a caller return address in the 'mem' node.
  * This will show the real owner of the 'mem' even it's allocated through wrapping APIs of malloc.
  */
-#define DEBUG_SET_CALLER_ADDR(mem) heapinfo_set_caller_addr(mem, __builtin_return_address(0))
+#define DEBUG_SET_CALLER_ADDR(mem) heapinfo_set_caller_addr(mem, (mmaddress_t)__builtin_return_address(0))
```

Comment:
@r-prabu @kishore-sn @abhishek-samsung We will change a type of the `mmaddress_t` from `size_t` to `void *`. Because it's an address.
```
#ifdef CONFIG_DEBUG_MM_HEAPINFO
typedef size_t mmaddress_t;             /* 32 bit address space */

->

typdef void *mmaddress_t;
```
How do you think?

Reply:
I think that would be good.